### DISCUSSION TITLE
Deleting Failed tasks to clear Airflow UI

### DISCUSSION THREAD

**Discussion by kennethpgreen** [apache/airflow#29189] [General]
I see in the best practices in airflow documentation that you should never delete a task for a DAG.

https://airflow.apache.org/docs/apache-airflow/2.0.0/best-practices.html#deleting-a-task

I've previously deleted tasks that I don't want to clear because I don't want them to rerun so I can clean up the Airflow UI and be more certain about which tasks are actually causing new errors that I haven't seen before. I'd like to follow Airflow best practices and not delete these tasks so I don't erase their history. Is their any recommended ways to avoid deleting tasks that you don't want to rerun but also maintain a more recent view in the airflow UI?

**Comment by potiuk** [comment]
Well. It really depends on what you want to achieve. If you want to stop seeing the taks and when you re run historical runs do not get it run, deleting taks is still ok. It will work, you will just not see not re-run the task in history.

I am not sure what you want to achieve by 'not rerunning' - if it is for historical runs then there is no good way to achieve it. If it is for new runs you could change conditions on the dependencies of your task so that it is always skipped for example - but you have to figure it out how to modify your DAGs to be always skipped and not affecting the overall DAG logic. 

But maybe you want to avhieve something else. Showing a specific example of what you want to achieve would be useful if you want some ideas how it can (or maybe cannot) be achieved.

**Reply by kennethpgreen** [reply]
Let's say I have a DAG that runs every hour. I care that this DAG stays up to date, but if a historical DAG run fails, I don't care about rerunning it. The DAG fails 1% of the time. If I'm looking at Airflow UI, I will likely see the red error for all the times that this DAG has failed in the past. In order to distinguish between old errors and new errors, I want to clean up any failures that have occurred prior to the most recent successful DAG run. 

I've previously deleted the tasks but I'm wondering if there's a better way to remove these errors without erasing them.

**Reply by potiuk** [reply]
I suggest you build a small custom dashboard for that using Airflow API (or even custom Plugin in Airflow). I think Airflow UI is specifically designed to show all information and trying to (ab)use it to do more logic or even worse - modifying and bending your DAG structure dynamically to achieve what you want is quite a bad idea in general.  Airlfow API is designed specifically for the purpose - if you want to present the data from historical runs in the way you find best, you should use the API to present it in the way you want.

It's likely a few hours Gig to use some easy-to-use UI tools to build the view using a logic of yours and Airlfow API. Likely less than problems and time needed to explore some workarounds that would (ab)use Airflow DAG structure changes for that.
